docs/source/demos/cost_benefit_matrix.ipynb

```diff
@@ -73,8 +73,11 @@
    "source": [
     "from evalml.demos.churn import load_churn\n",
     "from evalml.preprocessing import split_data\n",
+    "import pandas as pd\n",
+    "\n",
     "X, y = load_churn()\n",
     "X = X.set_types({'PaymentMethod':'Categorical', 'Contract': 'Categorical'}) # Update data types Woodwork did not correctly infer\n",
+    "y = pd.Series([0 if i == 'No' else 1 for i in y.to_series().values])\n",
```

Comment:
Since we're training the best pipeline and thresholding it, it requires the binary targets to be [0, 1]

Reply:
This feels like a pretty big (unintended) breaking change. If I'm following correctly, we can't use string-valued targets in `AutoMLSearch` if `train_best_pipeline=True` and `optimize_thresholds=True`?

Reply:
@freddyaboulton, this was really weird. Running it locally works as expected, like this
![image](https://user-images.githubusercontent.com/22552445/110383977-16d5ee00-802b-11eb-9cc6-4f7b9e8b2503.png)

But it fails when I try to make the docs. Not sure why it fails in both the docs and looking glass runs, but it throws an error in `objective.optimize_threshold` [here](https://github.com/alteryx/evalml/pull/1943/files#diff-d33b07734e238a4b877a6333e876204ecc236bd00d3a4549668e88bf6b3d802dR93)

Reply:
@bchen1116 and I dug into this and figured out it's because we need to encode the targets before `objective.optimize_threshold`. 

We discussed two implementations:
1. Calling `pipeline._encode_targets` in the call to `optimize_thresholds` like this:
```python
            pipeline.threshold = objective.optimize_threshold(y_predict_proba,
                                                              pipeline._encode_targets(y_threshold_tuning.to_series()),
                                                              X=X_threshold_tuning)
```
2. Making the threshold tuning a binary classfication pipeline method
```python
pipeline.tune_threshold(X, y_true, y_pred_proba, objective)
```

I prefer the second option because it's the pipeline's responsibility to encode targets before fit/score so I think the consistent thing to do would be to let the pipeline encode targets before threshold tuning. Calling `_encode_targets` outside a pipeline method doesn't feel right to me.